tests/contenttypes_tests/test_fields.py

```diff
@@ -27,9 +27,28 @@ def test_incorrect_get_prefetch_queryset_arguments(self):
 
 class GenericRelationTests(TestCase):
 
+    def setUp(self):
+        self.question = Question.objects.create(text='test')
+        self.answer1 = Answer.objects.create(question=self.question)
+        self.answer2 = Answer.objects.create(question=self.question)
+
     def test_value_to_string(self):
-        question = Question.objects.create(text='test')
-        answer1 = Answer.objects.create(question=question)
-        answer2 = Answer.objects.create(question=question)
-        result = json.loads(Question.answer_set.field.value_to_string(question))
-        self.assertCountEqual(result, [answer1.pk, answer2.pk])
+        result = json.loads(Question.answer_set.field.value_to_string(self.question))
+        self.assertCountEqual(result, [self.answer1.pk, self.answer2.pk])
+
+    def test_remove_deletes_related_object_from_database(self):
+        self.assertTrue(self.answer1 in Answer.objects.all())
+        self.assertTrue(self.answer2 in Answer.objects.all())
+
+        self.question.answer_set.remove(self.answer1)
+
+        self.assertFalse(self.answer1 in Answer.objects.all())
+        self.assertTrue(self.answer2 in Answer.objects.all())
```

Comment:
There is no need to check objects one by one (the same in all new tests):
```suggestion
        self.assertEqual(Answer.objects.all(), [self.answer2])
```

Reply:
Hi Felix,  I have seen these comments, I will work on them today. BTW thanks for the tips.